js.go

```diff
@@ -1662,6 +1662,7 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 			// after the AddConsumer returns.
 			if consumer == _EMPTY_ {
 				sub.jsi.consumer = info.Name
+				sub.jsi.nms = fmt.Sprintf(js.apiSubj(apiRequestNextT), stream, info.Name)
```

Comment:
We should probably do that under condition of `if isPullMode {}`.

Reply:
+1 setting the next message subject here would not have side effects for push consumers but better to guard it to be under the check for pull mode just in case.

Reply:
Added the guard.